api/v1beta1/tobiko_types.go

```diff
@@ -18,238 +18,160 @@ package v1beta1
 
 import (
 	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
-	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// Hash - struct to add hashes to status
-/*
-type Hash struct {
-	// Name of hash referencing the parameter
-	Name string `json:"name,omitempty"`
-	// Hash
-	Hash string `json:"hash,omitempty"`
-}
-*/
-
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // TobikoSpec defines the desired state of Tobiko
 type TobikoSpec struct {
-	// +kubebuilder:validation:Required
-	// +operator-sdk:csv:customresourcedefinitions:type=spec
-	// +kubebuilder:default="local-storage"
-        // StorageClass used to create PVCs that store the logs
-	StorageClass string `json:"storageClass"`
-
-        // +kubebuilder:validation:Optional
-	// +operator-sdk:csv:customresourcedefinitions:type=spec
-	// This value contains a nodeSelector value that is applied to test pods
-	// spawned by the test operator.
-	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	CommonOptions              `json:",inline"`
 
 	// +kubebuilder:validation:Optional
-        // +kubebuilder:default=false
-        // Activate debug mode. When debug mode is activated any error encountered
-        // inside the test-pod causes that the pod will be kept alive indefinitely
-        // (stuck in "Running" phase) or until the corresponding Tobiko CR is deleted.
-        // This allows the user to debug any potential troubles with `oc rsh`.
-        Debug bool `json:"debug"`
+	// +kubebuilder:default=false
+	// Activate debug mode. When debug mode is activated any error encountered
+	// inside the test-pod causes that the pod will be kept alive indefinitely
+	// (stuck in "Running" phase) or until the corresponding Tobiko CR is deleted.
+	// This allows the user to debug any potential troubles with `oc rsh`.
+	Debug bool `json:"debug"`
 
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
-	// This value contains a toleration that is applied to pods spawned by the
-	// test pods that are spawned by the test-operator.
-	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// +kubebuilder:default:="py3"
+	// Test environment
+	Testenv string `json:"testenv"`
 
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
-        // +kubebuilder:default:="py3"
```

Comment:
The majority of the linting fixes are not included in the `Fix linting` commit. Can we move these linting fixes into this commit? Maybe if you perform the linting fix as first commit it might make things easier for you? [1]

[1] https://github.com/openstack-k8s-operators/test-operator/pull/200/commits/2667eb3fe0b7d157ac0c94aa54204cd219019842